src/resspect/feature_handling_utils.py

```diff
@@ -1,16 +1,44 @@
+import io
 import logging
 import pandas as pd
+import tarfile
 
 def save_features(
         data: pd.DataFrame,
         location: str = "filesystem",
         filename: str = None,
 ):
+    """Save features from a pandas Dataframe."""
     if location == "filesystem":
         if filename is not None:
             data.to_csv(filename)
             logging.info("Features have been saved to: %s", filename)
         else:
             raise ValueError("filename must be provided if saving to the filesystem.")
     else:
-        raise NotImplementedError("Alternative storage method implementation tbd.")
\ No newline at end of file
+        raise NotImplementedError("Alternative storage method implementation tbd.")
+
+def load_external_features(
+        filename: str = None,
+        location: str = "filesystem",
+):
+    "Load features from a .csv file."
+    data = None
+    if location == "filesystem":
+        if filename is not None:
+            if '.tar.gz' in filename:
+                tar = tarfile.open(filename, 'r:gz')
```

Comment:
Not critical to do here, but I it looks like `tarfile` now offers a context mananger. It's probably the case that when this was first written, there wasn't one. So you should be able to do something like: 
```
with tarfile.open(filename, 'r:gz'):
    ...
```